packages/atomic/src/components/search/atomic-result-link/atomic-result-link.ts

```diff
@@ -0,0 +1,155 @@
+   *
+   * The template literal can reference any number of result properties from the parent result. It can also reference the window object.
+   *
+   * For example, the following markup generates an `href` value such as `http://uri.com?id=itemTitle`, using the result's `clickUri` and `itemtitle` fields.
+   * ```html
+   * <atomic-result-link href-template='${clickUri}?id=${raw.itemtitle}'></atomic-result-link>
+   * ```
+   */
+  @property({reflect: true, attribute: 'href-template'})
+  public hrefTemplate?: string;
+
+  @state() private linkAttributes?: Attr[];
+  @state() private stopPropagation?: boolean;
+  private removeLinkEventHandlers?: () => void;
+
+  private resultContext = createResultContextController(this);
+  private interactiveResultContext =
+    createInteractiveResultContextController(this);
+
+  @state() public bindings!: Bindings;
+  @state() public error!: Error;
+
+  public initialize() {
+    this.dispatchEvent(
+      buildCustomEvent(
+        'atomic/resolveStopPropagation',
+        (stopPropagation: boolean) => {
+          this.stopPropagation = stopPropagation;
+        }
+      )
+    );
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this.removeLinkEventHandlers) {
+      this.removeLinkEventHandlers();
+      this.removeLinkEventHandlers = undefined;
+    }
+  }
+
+  @bindingGuard()
+  @errorGuard()
+  render() {
+    return html`${when(this.result && this.interactiveResult, () => {
+      const result = this.result!;
+      const interactiveResult = this.interactiveResult!;
+
+      const href = isUndefined(this.hrefTemplate)
+        ? result.clickUri
+        : buildStringTemplateFromResult(
+            this.hrefTemplate,
+            result,
+            this.bindings
+          );
+
+      this.linkAttributes = getAttributesFromLinkSlotContent(
+        this,
+        'attributes'
+      );
+
+      return renderLinkWithItemAnalytics({
+        props: {
+          href,
+          onSelect: () => interactiveResult.select(),
+          onBeginDelayedSelect: () => interactiveResult.beginDelayedSelect(),
+          onCancelPendingSelect: () => interactiveResult.cancelPendingSelect(),
+          attributes: this.linkAttributes,
+          stopPropagation: this.stopPropagation,
+          className: 'link-style',
+          onInitializeLink: (cleanupCallback) => {
+            if (this.removeLinkEventHandlers) {
+              this.removeLinkEventHandlers();
+            }
+            this.removeLinkEventHandlers = cleanupCallback;
+          },
+        },
+      })(html`
+          ${this.renderDefaultSlotContent(
+            html`<atomic-result-text
+            field="title"
+            default="no-title"
+          ></atomic-result-text>`
+          )}
+        `);
+    })}`;
+  }
+
+  private get result(): AnyUnfoldedItem {
+    return this.resultContext.item as AnyUnfoldedItem;
```

Comment:
Another note here:

I usually don't like casts, but this one (and the one below in the interactiveResult getter) are actually OK (we have a "guard" in the render function that checks for null values, and FoldedResult extends Result, so it's all fine).

It would arguably be preferable to use a pattern similar to what we did in the atomic-result-text's initialize function, but it's really not worth dwelling on too much.